packages/@unimodules/core/android/src/main/java/org/unimodules/core/interfaces/ReactNativeHostHandler.kt

```diff
@@ -0,0 +1,32 @@
+package org.unimodules.core.interfaces
+
+import com.facebook.react.bridge.JavaScriptContextHolder
+import com.facebook.react.bridge.ReactApplicationContext
+
+interface ReactNativeHostHandler {
+  /**
+   * Given chance for modules to override react bundle file.
+   * e.g. for expo-updates
+   *
+   * @return custom path to bundle file, or null if not to override
+   */
+  fun getJSBundleFile(): String? {
+    return null
+  }
+
+  /**
+   * Given chance for modules to override react bundle asset name.
+   * e.g. for expo-updates
+   *
+   * @return custom bundle asset name, or null if not to override
+   */
+  fun getBundleAssetName(): String? {
+    return null
+  }
+
+  /**
+   * Given chance for JSI modules to register, e.g. for react-native-reanimated
+   */
+  fun onRegisterJSIModules(reactApplicationContext: ReactApplicationContext, jsContext: JavaScriptContextHolder) {
```

Comment:
that is specific for something like react-native-reanimated.
my proposal is to add the `createReactNativeHostHandlers` integration directly in react-native-reanimated.
however, this requires expo-modules, but expo-modules auto-linking  is based on react-native-cli auto-linking.

the question is: if a module both support cli auto-linking and expo-modules auto-linking, how could we deal with this case?

Reply:
share yet another experiment today:
gradle will not complain if a dependency be added twice, so it is okay for a module to be both react-native-cli auto-linking and expo-module auto-linking.

for react-native-reanimated and react-native-gesture-handler, we could have an expo-module adapter package only for   application listener hook but no native modules nor view managers.
something like this:

```java
public class ExpoReanimatedAdapterPackage extends BasePackage {
  @Override
  public List<? extends ReactNativeHostHandler> createReactNativeHostHandlers(Context context) {
    return Collections.singletonList(new ReactNativeHostHandler() {
      @Override
      public void onRegisterJSIModules(ReactApplicationContext reactApplicationContext, JavaScriptContextHolder jsContext) {
        new ReanimatedJSIModulePackage().getJSIModules(reactApplicationContext, reactApplicationContext);
      }
    });
  }
}
```

will follow this strategy if there is no concerns.

Reply:
> but expo-modules auto-linking is based on react-native-cli auto-linking

Other way around https://github.com/react-native-community/cli/issues/248#issue-422591744

> will follow this strategy if there is no concerns.

I don't see any immediate issues with this approach, looks fine.